### DISCUSSION TITLE
React-Select styling options and performance.

### DISCUSSION THREAD

**Discussion by crowd-dong** [JedWatson/react-select#4765] [Q&A]
Is there any performance gains between the different methods to style react-select?
Such as..

```JS
import { externalBlack, externalGreen } from "assets/styles/variables/color.scss"

const customStyles = {
  option: (provided, state) => ({
    ...provided,
    color: state.isSelected ? externalBlack : provided.color,
    backgroundColor: state.isSelected ? externalGreent  : null,
    cursor: 'pointer',
  }),
      
<Select
      styles={customStyle} 
        ....
    />
```

VS.

just providing a `classNamePrefix` and using a separate `scss` file to style.
Or is this just a matter of preference

**Comment by hermbo** [comment]
Probly best to use inline styles only when there is a boolean style change like your example. Everything else is best in a stylesheet when it comes to returning for later changes.
